code/data-tracker.js

```diff
@@ -371,5 +371,109 @@ $(document).on('knack-view-render.view_1996', function(event, scene) {
     });
 });
 
+$(document).on('knack-view-render.view_2357', function(event, page) {
+  //  now with minor changes, used for traffic count attachments field
+  //  this one affects the table that those with editing priviledges see
+  //  replace attachment filename with attachment type
+  //  find each attachment cell
+   $('td.field_3176').each(function() {
+       
+     //  find each attachment link within the cell
+      $(this).find('a').each(function( index ){
+
+        var attachmentType = '';
+      
+        //  search the neighboring field (attachmenty type) and retrieve the corresponding type 
+        $(this).closest('tr').children('td.field_3174').find('span').children('span').each(function( index2 ) {
+              if (index == index2) {
+                attachmentType = $(this).text();
+              }
+        });
+        
+        //  update link contents 
+        // and add a line break to make it consistent with the box next to it (BH)
+        $(this).html(attachmentType + "<br>");
+
+      });
+ 	});
+ });
```

Comment:
Hey @wchIV, the way you could edit this code to make it less copy/pasta is to put the repeated code blocks into a function. So, basically everything between 375-398 would become a function that you could call inside several views. See my example below:

```js
function updateAttachmentFilename() {
  //  now with minor changes, used for traffic count attachments field
  //  this one affects the table that those without editing priviledges see
  //  replace attachment filename with attachment type
  //  find each attachment cell
  $("td.field_3176").each(function() {
    //  find each attachment link within the cell
    $(this)
      .find("a")
      .each(function(index) {
        var attachmentType = "";

        //  search the neighboring field (attachmenty type) and retrieve the corresponding type
        $(this)
          .closest("tr")
          .children("td.field_3174")
          .find("span")
          .children("span")
          .each(function(index2) {
            if (index == index2) {
              attachmentType = $(this).text();
            }
          });

        //  update link contents
        // and add a line break to make it consistent with the box next to it (BH)
        $(this).html(attachmentType + "<br>");
      });
  });
}

$(document).on("knack-view-render.view_2357", function(event, page) {
  updateAttachmentFilename();
});

$(document).on("knack-view-render.view_2486", function(event, page) {
  updateAttachmentFilename();
});

```